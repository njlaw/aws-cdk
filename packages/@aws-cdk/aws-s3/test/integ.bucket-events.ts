import cdk = require('@aws-cdk/core');
import s3 = require('../lib');

const app = new cdk.App();
const stack = new cdk.Stack(app, 'aws-cdk-s3-events');

const bucket = new s3.Bucket(stack, 'Bucket');
// Use a raw CfnResource since aws-sns would introduce a circular dependency
const topic = new cdk.CfnResource(stack, 'MyTopic', {
    type: 'AWS::SNS::Topic',
});

bucket.onCloudTrailPutObject('OnPutObject', {
    target: {
        bind: () => ({
            arn: topic.ref,
            id: ''
        })
    },
    paths: [ 'key' ]
});

app.synth();
