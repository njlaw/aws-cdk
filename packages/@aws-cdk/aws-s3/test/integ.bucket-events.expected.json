{
  "Resources": {
    "Bucket83908E77": {
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
      "DeletionPolicy": "Retain"
    },
    "BucketOnPutObject88EA8A08": {
      "Type": "AWS::Events::Rule",
      "Properties": {
        "EventPattern": {
          "source": [
            "aws.s3"
          ],
          "detail-type": [
            "AWS API Call via CloudTrail"
          ],
          "detail": {
            "resources": {
              "ARN": [
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "Bucket83908E77",
                          "Arn"
                        ]
                      },
                      "/key"
                    ]
                  ]
                }
              ]
            },
            "eventName": [
              "CompleteMultipartUpload",
              "CopyObject",
              "PutObject"
            ],
            "requestParameters": {
              "bucketName": [
                {
                  "Ref": "Bucket83908E77"
                }
              ],
              "key": [
                "key"
              ]
            }
          }
        },
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Ref": "MyTopic"
            },
            "Id": "Target0"
          }
        ]
      }
    },
    "MyTopic": {
      "Type": "AWS::SNS::Topic"
    }
  }
}